### PR TITLE
Jz 5.10

### PR DESCRIPTION
**Pull request by citral23**
Pull request not for acceptance, just easier to track/merge for pcercuei I think.

KR

### CODE REVIEW

File: sound/soc/codecs/jz4770.c [hunk 1]
```diff
@@ -190,12 +190,12 @@ static int jz4770_codec_set_bias_level(struct snd_soc_component *codec,
 
 	switch (level) {
 	case SND_SOC_BIAS_PREPARE:
+		regmap_write(regmap, JZ4770_CODEC_REG_IFR, REG_IFR_ALL_MASK);
 		regmap_clear_bits(regmap, JZ4770_CODEC_REG_CR_VIC,
 				  REG_CR_VIC_SB);
 		msleep(250);
 		regmap_clear_bits(regmap, JZ4770_CODEC_REG_CR_VIC,
 				  REG_CR_VIC_SB_SLEEP);
-		msleep(400);
```
Comment (pcercuei) [inline sound/soc/codecs/jz4770.c:198]: Why is that one removed?

File: sound/soc/codecs/jz4770.c [hunk 2]
```diff
@@ -284,7 +284,7 @@ static int jz4770_codec_mute_stream(struct snd_soc_dai *dai, int mute, int direc
 		err = regmap_read_poll_timeout(jz_codec->regmap,
 					       JZ4770_CODEC_REG_IFR,
 					       val, val & gain_bit,
-					       1000, 100 * USEC_PER_MSEC);
+					       1000, 1000 * USEC_PER_MSEC);
```
Comment (pcercuei) [inline sound/soc/codecs/jz4770.c:287]: `1 * USEC_PER_SEC`

File: sound/soc/codecs/jz4770.c [hunk 3]
```diff
@@ -368,17 +368,22 @@ static int hpout_event(struct snd_soc_dapm_widget *w,
...
 	switch (event) {
 	case SND_SOC_DAPM_PRE_PMU:
-		/* set cap-less, unmute HP */
+		/* wait for the DAC to be ready after DAC_SB=0 */
+		msleep(400);
+
+		/* unmute HP */
 		regmap_clear_bits(jz_codec->regmap, JZ4770_CODEC_REG_CR_HP,
-				  REG_CR_HP_SB_HPCM | REG_CR_HP_MUTE);
+                                  REG_CR_HP_MUTE);
```
Comment (pcercuei) [inline sound/soc/codecs/jz4770.c:376]: Keep the same alignment - align with tabs, and complete with spaces so that it's aligned with the jz_codec->regmap

File: sound/soc/codecs/jz4770.c [hunk 4]
```diff
@@ -368,17 +368,22 @@ static int hpout_event(struct snd_soc_dapm_widget *w,
...
+
 		break;
 
 	case SND_SOC_DAPM_POST_PMU:
 		/* wait for ramp-up complete (RUP) */
 		err = regmap_read_poll_timeout(jz_codec->regmap,
 					       JZ4770_CODEC_REG_IFR,
 					       val, val & REG_IFR_RUP,
-					       1000, 100 * USEC_PER_MSEC);
+					       1000, 1000 * USEC_PER_MSEC);
```
Comment (pcercuei) [inline sound/soc/codecs/jz4770.c:386]: Same as above.

File: sound/soc/codecs/jz4770.c [hunk 5]
```diff
@@ -387,18 +392,17 @@ static int hpout_event(struct snd_soc_dapm_widget *w,
 		/* clear RUP flag */
 		regmap_set_bits(jz_codec->regmap, JZ4770_CODEC_REG_IFR,
 				REG_IFR_RUP);
-
```
Comment (pcercuei) [inline sound/soc/codecs/jz4770.c:390]: Unrelated change - shouldn't be in this patch.

File: sound/soc/codecs/jz4770.c [hunk 6]
```diff
@@ -387,18 +392,17 @@ static int hpout_event(struct snd_soc_dapm_widget *w,
...
+		/* mute HP */
 		regmap_set_bits(jz_codec->regmap, JZ4770_CODEC_REG_CR_HP,
-				REG_CR_HP_SB_HPCM | REG_CR_HP_MUTE);
+				REG_CR_HP_MUTE);
 
 		err = regmap_read_poll_timeout(jz_codec->regmap,
 					       JZ4770_CODEC_REG_IFR,
 					       val, val & REG_IFR_RDO,
-					       1000, 100 * USEC_PER_MSEC);
+					       1000, 1000 * USEC_PER_MSEC);
```
Comment (pcercuei) [inline sound/soc/codecs/jz4770.c:405]: Same as above.

File: sound/soc/codecs/jz4770.c [hunk 7]
```diff
@@ -636,8 +640,8 @@ static void jz4770_codec_codec_init_regs(struct snd_soc_component *codec)
 	/* Disable ADC lrswap */
 	regmap_set_bits(regmap, JZ4770_CODEC_REG_CR_ADC, REG_CR_ADC_LRSWAP);
 
-	/* default to cap-less mode(0) */
-	regmap_clear_bits(regmap, JZ4770_CODEC_REG_CR_HP, REG_CR_HP_SB_HPCM);
+	/* default to cap-coupled mode(1) */
+	regmap_set_bits(regmap, JZ4770_CODEC_REG_CR_HP, REG_CR_HP_SB_HPCM);
```
Comment (pcercuei) [inline sound/soc/codecs/jz4770.c:644]: Changing the default to cap-coupled will break the radio on the GCW-Zero. I think you should instead add a DAPM widget (switch, probably), that can then be linked in devicetree as part of the headphones/speakers route.

**Review by pcercuei — CHANGES_REQUESTED** [review]
You have three different changes in one patch, as suggested in the commit message. They should be in three different patches then. And start your commit titles with `ASoC: jz4770: `

File: sound/soc/codecs/jz4770.c (hunk 1 above)
Comment (citral23) [reply sound/soc/codecs/jz4770.c:198]: To be in line with the documentation recommendation : 
When SB_SLEEP falls, CODEC leaves the corresponding STANDBY mode; it is necessary to wait
some time before the CODEC reaches the normal mode. Depending on the selected mode, this time is
either called Tshd_adc (SB_ADC=0) for the ADC path or Tshd_dac (SB_DAC=0) for the DAC path.

Tshd_dac Typ 400 Max 900 is thus supposed to happen after SB_DAC=0 and isn't needed after SB_SLEEP=0 imho, this is simply subject to review I have no hard opinion.
